cves/cve_2021_41773.py

```diff
@@ -48,41 +47,79 @@
-    print(constants.FULL_QUESTION_MESSAGE.format('Is the filesystem directory in the configuration file set to "Require'
-                                                 ' all granted"?'))
-    if not start or not end:
-        print(constants.FULL_EXPLANATION_MESSAGE.format('Can not determine vulnerability status, no filesystem '
-                                                        'directory configuration value'))
-        return constants.UNSUPPORTED
-    for line in configuration_content[start:end]:
-        if 'Require all granted' in line:
-            print(constants.FULL_NEGATIVE_RESULT_MESSAGE.format('Yes'))
-            print(constants.FULL_EXPLANATION_MESSAGE.format('Apache configuration file sets the filesystem '
-                                                            'directory to "Require all granted"'))
-            return True
-        if 'Require all denied' in line:
-            print(constants.FULL_POSITIVE_RESULT_MESSAGE.format('No'))
-            print(constants.FULL_EXPLANATION_MESSAGE.format('Apache configuration file sets the filesystem '
-                                                            'directory to "Require all denied"'))
-            return False
-    print(constants.FULL_EXPLANATION_MESSAGE.format('Can not determine vulnerability status, no filesystem '
-                                                    'directory configuration value'))
-    return constants.UNSUPPORTED
+    cgi_bin_directory = None
+    directory_name = None
+    alias_module_section = False
+    directory_section = False
+    directories = []
+    cgi_bin_required_all_granted = False
+    print(constants.FULL_QUESTION_MESSAGE.format('Does the configuration have the vulnerable configuration set?\n1. '
+                                                 'The cgi-bin directory must be set as an alias module.\n2. The cgi-bin '
+                                                 'directory must be set to "Require all granted".\n3. At least another '
+                                                 'directory must be set to "Require all granted")'))
+    for line in configuration_content:
+        if line.startswith('<IfModule alias_module>'):
+            alias_module_section = True
+        elif line.startswith('</IfModule>'):
+            alias_module_section = False
+        if alias_module_section and 'ScriptAlias /cgi-bin/' in line:
+            cgi_bin_directory = line.split('ScriptAlias /cgi-bin/')[1]
+            if 'cgi-bin' in cgi_bin_directory:
+                cgi_bin_directory = parse_directory_name(cgi_bin_directory)
+        if line.startswith('<Directory'):
+            directory_section = True
+            directory_name = line.split('<Directory')[1].split('>')[0]
+            directory_name = parse_directory_name(directory_name)
+        if directory_section and 'Require all granted' in line:
+            if cgi_bin_directory and directory_name == cgi_bin_directory:
```

Comment:
Add parenthesis. It is currently unclear whether the logic you expect is: 
```
if (cgi_bin_directory and directory_name) == cgi_bin_directory
```
Or:
```
if cgi_bin_directory and (directory_name == cgi_bin_directory):
```